R/parallel.R

```diff
@@ -67,29 +67,42 @@ hipercow_parallel_get_cores <- function() {
 
 ##' Sets the environment variables `MC_CORES`, `OMP_NUM_THREADS`,
 ##' `OMP_THREAD_LIMIT`, `R_DATATABLE_NUM_THREADS` and `HIPERCOW_CORES` to
-##' the given number of cores. This is used to ensure that parallel workers you
-##' launch using [hipercow_parallel] will have the correct resources, but you
+##' the given number of cores. This is used to help various thread-capable
+##' packages use the correct number of cores. You
 ##' can also call it yourself if you know specifically how many cores you want
 ##' to be available to code that looks up these environment variables.
 ##'
 ##' @title Set various environment variables that report the number of cores
 ##' available for execution.
 ##'
+##' @inheritParams task_eval
+##'
 ##' @param cores Number of cores to be used.
 ##'
 ##' @export
-hipercow_parallel_set_cores <- function(cores) {
+hipercow_parallel_set_cores <- function(cores, envir = NULL) {
+  if (is.null(cores)) {
```

Comment:
I think this should not pass here; this is probably an error, no? So accept here a positive integer only?

Reply:
I'm getting a little confused by whether cores can be NULL or NA - I think it's probably ok to just take whatever hipercow_parallel_get_cores() returns (which is integer or NA, and we seem to be expecting a potential NA a couple of lines down...)...

See if this looks better now - I think the only place the NULL was coming from, was the new code I wrote, which I've now tamed below...